core/src/main/java/hudson/model/Job.java

```diff
@@ -1210,7 +1212,8 @@ public synchronized void doConfigSubmit(StaplerRequest req,
 
             String newName = req.getParameter("name");
             final ProjectNamingStrategy namingStrategy = Jenkins.getInstance().getProjectNamingStrategy();
-            if (newName != null && !newName.equals(name)) {
+            if (!StringUtils.equals(trim(newName), trim(name))) {
```

Comment:
trim() returns null for null input, and StringUtils.equals() handles nulls as desired here.  I can change this to trimNotNull, but I thought the point of adding the StringUtils methods here in the first place was to make the line more concise.